### ISSUE TITLE
Error: API calls being exceeded

### ISSUE THREAD

**Issue by rholligan** [signalkraft/mypyllant-component#204]
### Before submitting a new issue

- [x] I [redownloaded the latest release](https://my.home-assistant.io/redirect/hacs_repository/?owner=signalkraft&repository=mypyllant-component) and restarted Home Assistant
- [x] I've checked [known issues](https://signalkraft.com/mypyllant-component/#known-issues)
- [x] I've [enabled debug logs](https://signalkraft.com/mypyllant-component/3-contributing/#debugging) and attached the relevant logs to this ticket

### Problem description

For the last couple of days I'm getting this error relating to API calls being exceeded

### Logs

```
This error originated from a custom integration.

Logger: custom_components.mypyllant.coordinator
Source: helpers/update_coordinator.py:386
integration: myVAILLANT (documentation, issues)
First occurred: 25 August 2024 at 22:46:20 (9 occurrences)
Last logged: 12:41:04

Error fetching myVAILLANT data: Quota Exceeded on https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/homes, skipping update of myVAILLANT data for another 10800s
Error fetching myVAILLANT data: 403, message='Forbidden', url='https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/systems/xxx/meta-info/time-zone'
Error fetching myVAILLANT data: 403, message='Forbidden', url='https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/xxx/currentSystem'
Error fetching myVAILLANT data: 403, message='Forbidden', url='https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/systems/xxx/tli'
Error fetching myVAILLANT data: 504, message='Gateway Time-out', url='https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/systems/xxx/tli'
```

**Comment by rholligan** [comment]
Here are my settings relating to update frequency (unchanged for several months)

![Screenshot_20240826_212756_Home Assistant](https://github.com/user-attachments/assets/c7b68770-f46f-40bc-92a5-26e0519ca042)

**Comment by Krzysztonek** [comment]
Exactly same issue here.

**Comment by Boeroer** [comment]
I encountered the same problem yesterday. Interestingly it started after a technician finished the annual maintenance for the heat pump. But that must be a coincidence...?

**Comment by janezkranjc** [comment]
Started happening to me too, no technician and no update

**Comment by Telejunky** [comment]
I have the same problem since yesterday.

**Comment by Boeroer** [comment]
Vaillant send an eMail today and said that they updated their myVaillant app (it crashed when looking up the consuption data). Whatever they did to the app and most likely also the api, it fixed this integration for me, too. Fingers crossed!

**Comment by fodorg01** [comment]
@Boeroer : Thanks for the hint! I've just re-enabled the integration and it is now fetching the data, so it seems to be fine. Hopefully it stays like that and remains stable. Fingers crossed!🤞

**Comment by Boeroer** [comment]
My integration stopped working again. Same problem as before. The app works just fine (consumption data included).

**Comment by hzulla** [comment]
Same problem here.

![Bildschirmfoto vom 2024-10-25 19-16-37](https://github.com/user-attachments/assets/f873797c-f4aa-482a-98af-adebb98f0e42)

```
Error fetching myVAILLANT data: Quota Exceeded on https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/{id}/devices/{id}/buckets?resolution=HOUR&operationMode=COOLING&energyType=CONSUMED_ELECTRICAL_ENERGY&startDate=2024-10-24T00:00:00.000%2B02:00&endDate=2024-10-25T00:00:00.000%2B02:00, skipping update of myVAILLANT data for another 10752s
Error fetching myVAILLANT data: Quota Exceeded on https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/{id}/devices/{id}/buckets?resolution=HOUR&operationMode=COOLING&energyType=CONSUMED_ELECTRICAL_ENERGY&startDate=2024-10-25T00:00:00.000%2B02:00&endDate=2024-10-26T00:00:00.000%2B02:00, skipping update of myVAILLANT data for another 10800s
Error fetching myVAILLANT data: Quota Exceeded on https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/{id}/devices/{id}/buckets?resolution=HOUR&operationMode=COOLING&energyType=CONSUMED_ELECTRICAL_ENERGY&startDate=2024-10-25T00:00:00.000%2B02:00&endDate=2024-10-26T00:00:00.000%2B02:00, skipping update of myVAILLANT data for another 10744s
Error fetching myVAILLANT data: Quota Exceeded on https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/{id}/devices/{id}/buckets?resolution=HOUR&operationMode=COOLING&energyType=CONSUMED_ELECTRICAL_ENERGY&startDate=2024-10-25T00:00:00.000%2B02:00&endDate=2024-10-26T00:00:00.000%2B02:00, skipping update of myVAILLANT data for another 10726s
Error fetching myVAILLANT data: Quota Exceeded on https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/{id}/devices/{id}/buckets?resolution=HOUR&operationMode=COOLING&energyType=CONSUMED_ELECTRICAL_ENERGY&startDate=2024-10-25T00:00:00.000%2B02:00&endDate=2024-10-26T00:00:00.000%2B02:00, skipping update of myVAILLANT data for another 10717s
```

**Comment by hzulla** [comment]
> Seems like Vaillant is doing this on purpose.

It's possible to write to the app's official support address at app-support@vaillant-group.com

However, I'd recommend you contact your heating system installer (Heizungsbauer) and ask them to complain on your behalf to Vaillant. Your installer has a direct business contact relationship to Vaillant and the company is more likely to listen to customer complaints if they are reported through the installers.

(My installer gave me the contact address of his Vaillant Sales rep and the guy isn't just really helpful, he also gave me detailed technical advice when I had a problem with my Vaillant system's inclusion to my smart home setup where my installer couldn't help me.)

**Comment by Boeroer** [comment]
I currently only use the integration to ramp up the hot water temperature in times of plenty solar power. For that purpose I don't need to see data updates every 60 seconds (like the current default configuration suggests) because the automation is not triggered by data from the heat pump.
So I set the interval to 3630 seconds and reloaded the integration - and that seems to circumvent this one problem ("Quota exceeded") for now.
This is not ideal for lots of people of course, but for me it's enough atm. Because of that I won't bother my installer (yet).
But I think I will still write to the app-support and lay out the whole thing. At least they were responsive enough when I asked them about the consumption data bug.

**Comment by hzulla** [comment]
I'm really happy with my heating system and it's working fine. I recommend Vaillant to others who wa

However, it's remarkably silly that after paying a five digit sum for Vaillant gear there is no public documentation for the ebus protocol, the official ebus/KNX adapter is extremely limited in its feature set and the official app API gives me "quota exceeded" for benign 15 minute interval requests. It's not exotic in 2024 that home owners want to integrate their heat pump to their smart home system.

**Comment by Boeroer** [comment]
Aaand it's unavailable again, even with 3630 sec intervals. This is annoying as hell.

**Comment by Boeroer** [comment]
I did a hacky workaround since reloading the integration allows to fetch data at least once: I created an automation that reloads the integration once a particular sensor of the integration becomes unavailable for more than 1 minute. I know it's a dirty solution - but at least it works for now:

`- id: '1729964761593'
  alias: force myVaillant reload
  description: ''
  triggers:
  - trigger: state
    entity_id:
    - sensor.flexocompact_domestic_hot_water_0_tank_temperature
    to: unavailable
    for:
      hours: 0
      minutes: 1
      seconds: 0
  conditions: []
  actions:
  - action: homeassistant.reload_config_entry
    data:
      entry_id: '{{ config_entry_id(''sensor.flexocompact_domestic_hot_water_0_tank_temperature'')
        }}'
  mode: single`

**Comment by rholligan** [comment]
I only saw the issue between 26-Oct at ~12:00, and  at ~10:00
![image](https://github.com/user-attachments/assets/966bb89f-09a2-44b9-b0b5-8209412bf635)

**Comment by Boeroer** [comment]
The problem doesn't seem to be consistent.
My automation I talked about above which reloads the integration once its sensors become unavailable hasn't triggered since 9 hours.
Yesterday it triggered every 5 minutes (= the interval I configured to fetch new data).

**Comment by rholligan** [comment]
Yeah, almost as soon as I made the last post, my sensors became unavailable again 🥴

**Comment by mgiesl2s** [comment]
Is it possible to use an vpn or something else for home assistant or the the integration to change every request the external ip adress?

**Comment by mgiesl2s** [comment]
And why dont use this integration not the local eebus protocol over tcp/ip? SMA can for example connect via eebus locally to vaillant

**Comment by hzulla** [comment]
> Is it possible to use an vpn or something else for home assistant or the the integration to change every request the external ip adress?

You're using the API with your account login. If I were an API service provider, I'd use the quota per account, not per IP. Also, I'd consider actively bypassing a quota limit abusive.

As an API user, I want the API service provider to drop nonsensical quota limits instead of using clever tricks to bypass them.

> And why dont use this integration the local eebus protocol over tcp/ip? SMA can for example connect via eebus locally to vaillant

The Vaillant variant ebus protocol is undocumented. [ebusd](https://ebusd.de/) has reverse-engineered much of it and there is a [home assistant addon for ebusd adapters](https://github.com/LukasGrebe/ha-addons). (I'm using it in parallel to mypyllant.)

**Comment by mgiesl2s** [comment]
I doesnt wrote ebus. I wrote eebus. It is a protocol which communicates over tcp/ip. No ebusd is needet when using eebus instead. But i already doesnt find an eebus integration for home assistant. So it will be time to create one :-) who can this?

**Comment by Boeroer** [comment]
Some Vaillant communication modules like the VR 900 don't support eebus (only VR 920), although it was promised that they'd come eebus-ready.
However, they work with the myVaillant app and -API.

**Comment by boelle** [comment]
@rholligan is this still an issue?

**Comment by boelle** [comment]
if its stays that way then maybe close the issue?

i'm just trying to help @signalkraft getting the list shorter with only problems still present

**Comment by water-escape** [comment]
Anyone found the correct settings?
I just got Error fetching myVAILLANT data: Quota Exceeded again

i have set:

360
4200
1
365
15
30

Fetch real-time power usage (not supported on every system)
Fetch Energy Management Data

**Comment by wolfma61** [comment]
I haven't had myVaillant quota problems for over a month now.
Using these settings:

Seconds between updates 7200
seconds between energy data 7200
delay 60

fetching real-time statistics, power usage, connection status, trouble code, energy management data is turned on. 

you can probably tweak further, but this is detailed enough for me ...

I installed a ShellyEM4 for energy measurement on the power connection of the Vaillant heat pump. That gives me accurate power consumption information. It turns out, that the power consumption reported by myVaillant and Shelly are very close.

Wolfgang

**Comment by Krzysztonek** [comment]
And here are my settings, which work fine for me since Nov'2024:

<img width="566" height="635" alt="Image" src="https://github.com/user-attachments/assets/443382d6-dc20-4b86-adfa-7202f12f6198" />

No quota issues have been faced since then.
